amazon_product_widget.module

```diff
@@ -110,3 +114,71 @@ function amazon_product_widget_get_all_asins() {
   }
   return $asins;
 }
+
+/**
+ * Called when cron is run.
+ */
+function amazon_product_widget_cron_update_deals() {
+  /** @var \Drupal\Core\State\StateInterface $state */
+  $state = \Drupal::state();
+  /** @var \Drupal\amazon_product_widget\DealFeedService $dealFeedService */
+  $dealFeedService = \Drupal::service('amazon_product_widget.deal_feed_service');
+
+  // Is there a process already running?
+  $running = $state->get('amazon_product_widget.deal_cron_running', FALSE);
+  if (!$running) {
+    $currentTime = time();
+    $lastRun = $state->get('amazon_product_widget.deal_cron_last_run', 0);
+    $nextRun = $lastRun + $dealFeedService->getCronInterval() * 60;
+
+    if ($currentTime < $nextRun) {
+      return;
+    }
+
+    $importFilename = NULL;
+    try {
+      $importFilename = $dealFeedService->downloadDealsCsv();
+    }
+    catch (\Throwable $exception) {
+      watchdog_exception('amazon_product_widget', $exception);
+      return;
+    }
+
+    $state->set('amazon_product_widget.deal_cron_running', TRUE);
+    $dealsCronState = [
+      'file' => $importFilename,
+      'processed' => 0,
+      'errors' => 0,
+    ];
+    $state->set('amazon_product_widget.deal_cron_state', $dealsCronState);
+    goto startCron;
+  }
+  else {
+    startCron:
+
+    $importState = $state->get('amazon_product_widget.deal_cron_state');
+    // We allow more imports per round when updating via cron.
+    $importsPerRound = 10000;
+    try {
+      $errors = $dealFeedService->import(
+        $importState['file'],
+        $importState['processed'],
+        $importsPerRound
+      );
+
+      $importState['processed'] += $importsPerRound;
+      $importState['errors'] += $errors;
+
+      if ($importState['errors'] >= $dealFeedService->getMaxDealImportErrors()) {
+        throw new DealFeedFinishedProcessingException();
+      }
+    }
+    catch (DealFeedFinishedProcessingException $exception) {
+      $state->set('amazon_product_widget.deal_cron_running', FALSE);
+      $state->set('amazon_product_widget.deal_cron_last_run', time());
+      $state->delete('amazon_product_widget.deal_cron_state');
```

Comment:
do we need some kind of reset like this at the start of the cronjob as well, in case the state contains stuff from a cancelled cronjob or something